### PR TITLE
add birthdate validator

### PR DESCRIPTION
**Pull request by siyu7866**
Add birthdate & username validators. I did not merge submit button and login service, because I do not know what kind of data will be passed through this button.

### CODE REVIEW

File: src/app/auth/login/login.page.ts [hunk 1]
```diff
@@ -39,4 +61,21 @@ export class LoginPage implements OnInit {
     this.authService.logout();
   }
 
+    // validation message when input information is incorrect
+    validation_messages = {
```
Comment (nash90) [inline src/app/auth/login/login.page.ts:65]: Please match the indentation to previous code block. Also move the declaration to before constructor.

File: src/app/validators/username.validator.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { FormControl } from '@angular/forms';
+
+export class UsernameValidator {
+
+  static validUsername(fc: FormControl){
+
+    if(fc.value.toLowerCase() === "abc123" || fc.value.toLowerCase() === "123abc"){
```
Comment (nash90) [inline src/app/validators/username.validator.ts:7]: Is this validation being used? If not please remove.

**Comment by nash90** [comment]
image

This issue is still there. login asking for full date in UI

File: src/app/auth/login/login.page.ts (hunk 1 above)
Comment (nash90) [reply src/app/auth/login/login.page.ts:65]: There should be automatic code formatting check if you use visual studio code. there is linter file in the project which looks after code formatting.

Comment (siyu7866) [reply src/app/auth/login/login.page.ts:65]: Thanks for your comments and information. I will check and modify it.

**Comment by siyu7866** [comment]
Year input in login layout should be in YYYY format. 
屏幕快照 2019-09-26 下午2 41 30
If you find anything different, please let me know.

**Comment by nash90** [comment]
> Year input in login layout should be in YYYY format.
> 屏幕快照 2019-09-26 下午2 41 30
> If you find anything different, please let me know.

I am using google chrome (iphone x) simulator. I get this: 
image

Can you let me know which browser you are testing?

**Comment by siyu7866** [comment]
I understand what happened. Because I set input type for date, there will be a date picker for that on phone. I am using safari for testing. I will check again, and modify it.

**Comment by nash90** [comment]
Code is merged and fixed couple of things

**Pull request closed without merging**